prusti-server/src/process_verification.rs

```diff
@@ -120,11 +266,129 @@ pub fn process_verification_request<'v, 't: 'v>(
+            match jni.class_name(msg).as_str() {
+                "viper.silver.reporter.QuantifierInstantiationsMessage" => {
+                    let msg_wrapper = silver::reporter::QuantifierInstantiationsMessage::with(env);
+                    let q_name =
+                        jni.get_string(jni.unwrap_result(msg_wrapper.call_quantifier(msg)));
+                    let q_inst = jni.unwrap_result(msg_wrapper.call_instantiations(msg));
+                    debug!("QuantifierInstantiationsMessage: {} {}", q_name, q_inst);
+                    // also matches the "-aux" and "_precondition" quantifiers generated
+                    // we encoded the position id in the line and column number since this is not used by
+                    // prusti either way
+                    if q_name.starts_with("prog.l") {
+                        let no_pref = q_name.strip_prefix("prog.l").unwrap();
+                        let stripped = no_pref
+                            .strip_suffix("-aux")
+                            .or(no_pref.strip_suffix("_precondition"))
+                            .unwrap_or(no_pref);
+                        let parsed = stripped.parse::<u64>();
+                        match parsed {
+                            Ok(pos_id) => {
+                                sender
+                                    .send(ServerMessage::QuantifierInstantiation {
+                                        q_name,
+                                        insts: u64::try_from(q_inst).unwrap(),
+                                        pos_id,
+                                    })
+                                    .unwrap();
+                            }
+                            _ => info!("Unexpected quantifier name {}", q_name),
+                        }
+                    }
+                }
+                "viper.silver.reporter.QuantifierChosenTriggersMessage" => {
+                    let obj_wrapper = java::lang::Object::with(env);
+                    let positioned_wrapper = silver::ast::Positioned::with(env);
+                    let msg_wrapper = silver::reporter::QuantifierChosenTriggersMessage::with(env);
+
+                    let viper_quant = jni.unwrap_result(msg_wrapper.call_quantifier(msg));
+                    let viper_quant_str =
+                        jni.get_string(jni.unwrap_result(obj_wrapper.call_toString(viper_quant)));
+                    // we encoded the position id in the line and column number since this is not used by
+                    // prusti either way
+                    let pos = jni.unwrap_result(positioned_wrapper.call_pos(viper_quant));
+                    let pos_string =
+                        jni.get_string(jni.unwrap_result(obj_wrapper.call_toString(pos)));
+                    let pos_id_index = pos_string.rfind('.').unwrap();
+                    let pos_id = pos_string[pos_id_index + 1..].parse::<u64>().unwrap();
```

Comment:
Why is the position handling different here and in the quantifier instantiations message? Can they share code?

Reply:
The handling is different due to the different message types. In the QuantifierInstantiationsMessage, we get a Z3 quantifier name (e.g. a string `prog.l234-aux`) while in the QuantifierChosenTriggersMessage, we get an actual `Positioned` jObject, where we extract the position id from the string of the position (because we needed less Java boiler plate code, but maybe it would be better to get it from the actual line member).